src/catalog.c

```diff
@@ -224,75 +216,111 @@ catalog_owner(void)
 	return owner_oid;
 }
 
-Catalog *
-catalog_get(void)
+static const char *
+catalog_table_name(CatalogTable table)
 {
-	int			i;
+	return catalog_table_names[table].table_name;
+}
 
-	if (!OidIsValid(MyDatabaseId))
-		elog(ERROR, "invalid database ID");
+void
+catalog_database_info_init(CatalogDatabaseInfo *info)
+{
+	info->database_id = MyDatabaseId;
+	StrNCpy(info->database_name, get_database_name(MyDatabaseId), NAMEDATALEN);
+	info->schema_id = get_namespace_oid(CATALOG_SCHEMA_NAME, false);
+	info->owner_uid = catalog_owner();
 
-	if (MyDatabaseId == catalog.database_id)
-		return &catalog;
+	if (info->schema_id == InvalidOid)
+		elog(ERROR, "OID lookup failed for schema \"%s\"", CATALOG_SCHEMA_NAME);
+}
 
-	if (!extension_is_loaded() || !IsTransactionState())
-		return &catalog;
+CatalogDatabaseInfo *
+catalog_database_info_get()
+{
+	if (!OidIsValid(database_info.database_id))
```

Comment:
should this have the same kind of extension installed checks as catalog_get()? But maybe we throw errors if extension not installed?

Reply:
Oh woops, good point. OK, I error if the extension isn't installed, and modified `catalog_get` to also error.